edgelet/aziot-edged/src/lib.rs

```diff
@@ -499,12 +490,10 @@ where
         settings.hostname().to_string().to_lowercase(),
     );
 
-    if let Some(parent_hostname) = settings.parent_hostname() {
```

Comment:
We want to modify (not remove) the check to compare the parent_hostname to hostname. If they are the same, we should skip inserting the `GATEWAY_HOSTNAME_KEY` environment var.

Reply:
parent_hostname == hostname is a possible use case?